### PR TITLE
analyze: several systemd-analyze plot improvements

### LINKED ISSUE #9662: RFE: Save service start/running time info from the initrd for systemd-analyze
Currently even if the initrd is using systemtd, as it is e.g on Fedora, the output of systemd-analyze plot will only contain a single long bar for the initrd.

It would be nice if on pivot to the real root systemd could save the statistics from the initrd, so that both initrd and normal services are individually shown  in systemd-analyze plot output.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #9662.

### CODE REVIEW

File: src/core/main.c [hunk 1]
```diff
@@ -2373,8 +2373,13 @@ int main(int argc, char *argv[]) {
...
         m->timestamps[MANAGER_TIMESTAMP_INITRD] = initrd_timestamp;
         m->timestamps[MANAGER_TIMESTAMP_USERSPACE] = userspace_timestamp;
-        m->timestamps[MANAGER_TIMESTAMP_SECURITY_START] = security_start_timestamp;
-        m->timestamps[MANAGER_TIMESTAMP_SECURITY_FINISH] = security_finish_timestamp;
+        if (in_initrd()) {
+                m->timestamps[MANAGER_TIMESTAMP_INITRD_SECURITY_START] = security_start_timestamp;
+                m->timestamps[MANAGER_TIMESTAMP_INITRD_SECURITY_FINISH] = security_finish_timestamp;
+        } else {
+                m->timestamps[MANAGER_TIMESTAMP_SECURITY_START] = security_start_timestamp;
+                m->timestamps[MANAGER_TIMESTAMP_SECURITY_FINISH] = security_finish_timestamp;
```
Comment (poettering) [inline src/core/main.c]: i wonder if we can find a nicer syntax for this. maybe add a new function like this:

```c
static ManagerTimestamp manager_timestamp_initrd_mangle(ManagerTimestamp s) {
    if (in_initrd() &&
            s >= MANAGER_TIMESTAMP_SECURITY_START &&
            s <= MANAGER_TIMESTAMP_UNITS_LOAD_FINISH)
                return s - MANAGER_TIMESTAMP_SECURITY_START + MANAGER_TIMESTAMP_INITRD_SECURITY_START;
        return s;
}
```

This basically maps the non-initrd version of these timestamp ids to the initrd version. We could then simply wrap the array index in the code above in a call to this function, and things would be a bit prettier I think

**Comment by poettering** [comment]
looks pretty good to me, just one comment

**Comment by yuwata** [comment]
@poettering Thank you for the review. I've added the helper function and use it where applicable.

**Comment by poettering** [comment]
lgtm

**Pull request merged**